pkg/resourcemanager/keyvaults/keyops.go

```diff
@@ -81,6 +81,24 @@ func (k *KeyvaultKeyClient) Ensure(ctx context.Context, obj runtime.Object, opts
 		KeyOps:        &instance.Spec.Operations,
 		KeyAttributes: &katts,
 	}
+
+	switch instance.Spec.Type {
+	case kvops.RSA, kvops.RSAHSM:
+		if instance.Spec.KeySize == 0 {
+			instance.Status.Message = "no keysize provided for rsa key"
+			instance.Status.FailedProvisioning = true
```

Comment:
Do you need to also set `instance.Status.Provisioning = false` here (and below)?